source/extensions/filters/http/on_demand/on_demand_update.cc

```diff
@@ -3,18 +3,118 @@
 #include "source/common/common/assert.h"
 #include "source/common/common/enum_to_int.h"
 #include "source/common/common/logger.h"
+#include "source/common/config/xds_resource.h"
 #include "source/common/http/codes.h"
+#include "source/common/http/utility.h"
+#include "source/extensions/filters/http/well_known_names.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace OnDemand {
 
-Http::FilterHeadersStatus OnDemandRouteUpdate::decodeHeaders(Http::RequestHeaderMap&, bool) {
+namespace {
+
+class RdsDecodeHeadersBehavior : public DecodeHeadersBehavior {
+public:
+  void decodeHeaders(OnDemandRouteUpdate& filter) override { filter.handleMissingRoute(); }
+};
+
+class RdsCdsDecodeHeadersBehavior : public DecodeHeadersBehavior {
+public:
+  RdsCdsDecodeHeadersBehavior(Upstream::OdCdsApiHandlePtr odcds, std::chrono::milliseconds timeout)
+      : odcds_(std::move(odcds)), timeout_(timeout) {}
+
+  void decodeHeaders(OnDemandRouteUpdate& filter) override {
+    auto route = filter.handleMissingRoute();
+    if (!route.has_value()) {
+      return;
```

Comment:
This is missing coverage.

Reply:
Argh, yes. It became a separate code path after adding the behavior classes.

Reply:
Covered. The only uncovered line is the locator stuff, but I don't know how to meaningfully test it.